In [2]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt

Lets import the csv file of the final coded responses. Remember that this is the final file along with all the analysis that we will try to do during the courses of this project. We will compare our outputs to the ones in this file.

In [30]:
responses = pd.read_csv('coded_responses.csv')
responses.head()

Unnamed: 0     id  gender  gender_coded     batch  batch_coded  \
0           0  17/20  Female             1  Final yr            1   
1           1   17/1  Female             1  Final yr            1   
2           2  17/10  Female             1  Final yr            1   
3           3  17/11  Female             1  Final yr            1   
4           4  17/12  Female             1  Final yr            1   

          avg_internet_usage  avg_internet_usage_coded             usage_type  \
0       6 to 8 hours per day                         4  Online lecture Marrow   
1       4 to 6 hours per day                         3        Online learning   
2  more than 8 hours per day                         5        Online learning   
3       6 to 8 hours per day                         4        Online learning   
4       6 to 8 hours per day                         4        Online learning   

   usage_type_coded  ...  environment_score_raw  environment_score  \
0                 8  ...                     23               11.5   
1                 1  ...                     40               20.0   
2                 1  ...                     22               11.0   
3                 1  ...                     28               14.0   
4                 1  ...                     26               13.0   

   environment_domain_round  environment_domain_range      avg_pre_covid_use  \
0                        12                      9-12     4 to 6 hrs per day   
1                        20                     17-20  8 or more hrs per day   
2                        11                      9-12     6 to 8 hrs per day   
3                        14                     13-16     2 to 4 hrs per day   
4                        13                     13-16     4 to 6 hrs per day   

   avg_pre_covid_use_coded     avg_post_covid_use  avg_post_covid_use_coded  \
0                        3     6 to 8 hrs per day                         4   
1                        5  8 or more hrs per day                         5   
2                        4     4 to 6 hrs per day                         3   
3                        2  8 or more hrs per day                         5   
4                        3     6 to 8 hrs per day                         4   

   change_in_usage  change_in_usage_coded  
0        INCREASED                      1  
1        NO CHANGE                      3  
2        DECREASED                      2  
3        INCREASED                      1  
4        INCREASED                      1  

[5 rows x 105 columns]

In [31]:
responses.shape

(194, 105)

So we have the original 194 responses and 105 columns. but the first column is just a previous index. we can drop it.

In [32]:
responses = responses.drop(columns='Unnamed: 0')

In [33]:
responses.head()

id  gender  gender_coded     batch  batch_coded  \
0  17/20  Female             1  Final yr            1   
1   17/1  Female             1  Final yr            1   
2  17/10  Female             1  Final yr            1   
3  17/11  Female             1  Final yr            1   
4  17/12  Female             1  Final yr            1   

          avg_internet_usage  avg_internet_usage_coded             usage_type  \
0       6 to 8 hours per day                         4  Online lecture Marrow   
1       4 to 6 hours per day                         3        Online learning   
2  more than 8 hours per day                         5        Online learning   
3       6 to 8 hours per day                         4        Online learning   
4       6 to 8 hours per day                         4        Online learning   

   usage_type_coded  iat_1  ...  environment_score_raw  environment_score  \
0                 8      2  ...                     23               11.5   
1                 1      3  ...                     40               20.0   
2                 1      3  ...                     22               11.0   
3                 1      2  ...                     28               14.0   
4                 1      4  ...                     26               13.0   

   environment_domain_round  environment_domain_range      avg_pre_covid_use  \
0                        12                      9-12     4 to 6 hrs per day   
1                        20                     17-20  8 or more hrs per day   
2                        11                      9-12     6 to 8 hrs per day   
3                        14                     13-16     2 to 4 hrs per day   
4                        13                     13-16     4 to 6 hrs per day   

   avg_pre_covid_use_coded     avg_post_covid_use  avg_post_covid_use_coded  \
0                        3     6 to 8 hrs per day                         4   
1                        5  8 or more hrs per day                         5   
2                        4     4 to 6 hrs per day                         3   
3                        2  8 or more hrs per day                         5   
4                        3     6 to 8 hrs per day                         4   

   change_in_usage  change_in_usage_coded  
0        INCREASED                      1  
1        NO CHANGE                      3  
2        DECREASED                      2  
3        INCREASED                      1  
4        INCREASED                      1  

[5 rows x 104 columns]

In [34]:
responses.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 194 entries, 0 to 193
Columns: 104 entries, id to change_in_usage_coded
dtypes: float64(11), int64(73), object(20)
memory usage: 157.8+ KB


if you look closely a lot of these columns are just a code to convert a categorical column to the numerical one. we can drop the codes for now. 

In [35]:
df = responses.copy()

In [36]:
df.columns

Index(['id', 'gender', 'gender_coded', 'batch', 'batch_coded',
       'avg_internet_usage', 'avg_internet_usage_coded', 'usage_type',
       'usage_type_coded', 'iat_1',
       ...
       'environment_score_raw', 'environment_score',
       'environment_domain_round', 'environment_domain_range',
       'avg_pre_covid_use', 'avg_pre_covid_use_coded', 'avg_post_covid_use',
       'avg_post_covid_use_coded', 'change_in_usage', 'change_in_usage_coded'],
      dtype='object', length=104)

In [37]:
df = df.drop(columns=['gender_coded', 'batch_coded', 'avg_internet_usage_coded', 'usage_type_coded',
                      'iat_range_coded', 'qol_range_coded', 'avg_pre_covid_use_coded', 
                      'avg_post_covid_use_coded', 'change_in_usage_coded'])
df.shape

(194, 95)

so we are already down a few columns

In [38]:
df.head()

id  gender     batch         avg_internet_usage             usage_type  \
0  17/20  Female  Final yr       6 to 8 hours per day  Online lecture Marrow   
1   17/1  Female  Final yr       4 to 6 hours per day        Online learning   
2  17/10  Female  Final yr  more than 8 hours per day        Online learning   
3  17/11  Female  Final yr       6 to 8 hours per day        Online learning   
4  17/12  Female  Final yr       6 to 8 hours per day        Online learning   

   iat_1  iat_2  iat_3  iat_4  iat_5  ...  social_score  social_domain_round  \
0      2      2      0      1      2  ...     10.666667                   11   
1      3      3      0      0      1  ...      9.333333                    9   
2      3      1      0      0      1  ...     16.000000                   16   
3      2      0      0      0      1  ...      8.000000                    8   
4      4      3      0      1      4  ...     14.666667                   15   

   social_domain_range  environment_score_raw  environment_score  \
0                 9-12                     23               11.5   
1                 9-12                     40               20.0   
2                13-16                     22               11.0   
3                  4-8                     28               14.0   
4                13-16                     26               13.0   

   environment_domain_round  environment_domain_range      avg_pre_covid_use  \
0                        12                      9-12     4 to 6 hrs per day   
1                        20                     17-20  8 or more hrs per day   
2                        11                      9-12     6 to 8 hrs per day   
3                        14                     13-16     2 to 4 hrs per day   
4                        13                     13-16     4 to 6 hrs per day   

      avg_post_covid_use  change_in_usage  
0     6 to 8 hrs per day        INCREASED  
1  8 or more hrs per day        NO CHANGE  
2     4 to 6 hrs per day        DECREASED  
3  8 or more hrs per day        INCREASED  
4     6 to 8 hrs per day        INCREASED  

[5 rows x 95 columns]

Note that i collected this data and was very careful with spellings and punctuation of entries from the collection stage itself. so we don't need to clean entry errors. but values are a lot more verbose than we need for analysis, as were the column names and ids that we cleaned in the process script. so lets simplify them.

In [39]:
df['gender'].value_counts()

gender
Male      117
Female     77
Name: count, dtype: int64

In [40]:
gender_dict = {'Female': 'F', 'Male': 'M'}
df['gender'] = df['gender'].apply(lambda x: gender_dict[x])

In [41]:
df['gender'].head()

0    F
1    F
2    F
3    F
4    F
Name: gender, dtype: object

In [42]:
df['batch'].value_counts()

batch
First yr    139
Final yr     55
Name: count, dtype: int64

In [43]:
batch_dict = {'First yr': 'junior',
              'Final yr': 'senior'}
df['batch'] = df['batch'].apply(lambda x: batch_dict[x])

In [44]:
df['batch'].head()

0    senior
1    senior
2    senior
3    senior
4    senior
Name: batch, dtype: object

In [45]:
df['avg_internet_usage'].value_counts()

avg_internet_usage
4 to 6 hours per day         61
2 to 4 hours per day         58
6 to 8 hours per day         30
0 to 2 hours per day         28
more than 8 hours per day    17
Name: count, dtype: int64

Pretty clean and good. no need to do anything with these.

In [46]:
df['usage_type'].value_counts()

usage_type
Social media                                        86
Online learning                                     38
Streaming services/ Entertainment                   20
Information gathering for academic purposes         17
Browsing                                            11
Communication                                        8
Online Gaming                                        7
Online lecture Marrow                                1
Dhruv rathi videos                                   1
Marrow and Netflix                                   1
Lecture videos                                       1
Study                                                1
For academic and social media,  watching series      1
Watching videos for study                            1
Name: count, dtype: int64

now there is a lot of overlap between categories, primarily because the form that i used, had a text input option if you couldn't find your answer and well, if you let people write, they get creative. so let's do something about this. i'll only change the entries that already overlap with the responses i gave in the form as options.

In [47]:
usage_type_dict = {"Online lecture Marrow": "Online learning",
                   "Dhruv rathi videos": "Browsing",
                   "Lecture videos": "Online learning",
                   "Study": "Online learning",
                   "Watching videos for study": "Online learning",
                   "Marrow and Netflix": "More than 1 of the above",
                   "For academic and social media,  watching series ": "More than 1 of the above"}

df['usage_type'] = df['usage_type'].apply(lambda x: usage_type_dict[x] if x in usage_type_dict.keys() else x)

In [48]:
df['usage_type'].value_counts()

usage_type
Social media                                   86
Online learning                                42
Streaming services/ Entertainment              20
Information gathering for academic purposes    17
Browsing                                       12
Communication                                   8
Online Gaming                                   7
More than 1 of the above                        2
Name: count, dtype: int64

okay thats better!

Now the Original questionaire had three types of questions. Questions from the IAT questionaaire, questions from the WHO-QOL questionaiire (i'll explain these as we work with them.) and questions about basic internet usage and demographic information.So we will separate the dataframe into three dataframes since these three domains can be analysed separately.

In [49]:
basics_df = df[['id','gender', 'batch', 'avg_internet_usage', 'usage_type', 'avg_pre_covid_use',
               'avg_post_covid_use', 'change_in_usage']]
basics_df.head()

id gender   batch         avg_internet_usage       usage_type  \
0  17/20      F  senior       6 to 8 hours per day  Online learning   
1   17/1      F  senior       4 to 6 hours per day  Online learning   
2  17/10      F  senior  more than 8 hours per day  Online learning   
3  17/11      F  senior       6 to 8 hours per day  Online learning   
4  17/12      F  senior       6 to 8 hours per day  Online learning   

       avg_pre_covid_use     avg_post_covid_use change_in_usage  
0     4 to 6 hrs per day     6 to 8 hrs per day       INCREASED  
1  8 or more hrs per day  8 or more hrs per day       NO CHANGE  
2     6 to 8 hrs per day     4 to 6 hrs per day       DECREASED  
3     2 to 4 hrs per day  8 or more hrs per day       INCREASED  
4     4 to 6 hrs per day     6 to 8 hrs per day       INCREASED

In [50]:
basics_df.isna().sum()

id                    0
gender                0
batch                 0
avg_internet_usage    0
usage_type            0
avg_pre_covid_use     0
avg_post_covid_use    0
change_in_usage       0
dtype: int64

you might have noticed that i didn't do the null value check on the original df. thats because there are 7 columns with null values but they are part of the analysis, and we will get to them later.

there are no null values in the collected data because the participitants who consented for the study had a form where all questions were mandatory.

that is also the reason we only have full data for 194 out of 350 possible participants. either the participants didn't fill the form at all or they filled it completely because incomplete forms wouldn't be submitted. this was a design choice.

iat dataframe with only the responses.

In [52]:
# these are the raw responses i had
iat_columns =['id']
for column in df.columns:
    if 'iat' in column:
        iat_columns.append(column)

iat_columns = iat_columns[:-2]
len(iat_columns)

21

In [53]:
iat_df = df[iat_columns]
iat_df.shape

(194, 21)

qol scores without the subdomain analysis

In [54]:
# these are the raw responses i had
qol_columns =['id']
for column in df.columns:
    if 'qol' in column:
        qol_columns.append(column)

qol_columns = qol_columns[:-2]
len(qol_columns)

30

if you look at the list of qol columns, you'll see that 3 questions have 2 columns each. one a response and the other the reversed value. that is a part of analysis too because of the structure of the QOL questionaaire. we will drop them too. (i'll explain the reason for the reversal when we analyse the dataframe.)

In [55]:
qol_df = df[qol_columns]
qol_df = qol_df.drop(columns=['qol_3_rev', 'qol_4_rev', 'qol_26_rev'])
qol_df.shape

(194, 27)

so now we have a basic df, an iat df (20 questions) and a qol df(26 questions). its sufficiently clean, due to care in collection process and now we can get on with the analysis.Let's export these dataframes to csv so we can use them in other notebooks.

run the following cell only once.

In [56]:
basics_df.to_csv('basics.csv')
iat_df.to_csv('iat_responses.csv')
qol_df.to_csv('qol_responses.csv')

Now lets get down to analysis. 